server/src/graphql/mutations/createAppGithub.ts

```diff
@@ -34,37 +40,38 @@ export const createAppGithub: MutationResolvers['createAppGithub'] = async (
 
   const octokit = new Octokit({});
 
-  let repo;
-
   const repoData = getRepoData(input.gitRepoUrl);
-  try {
-    repo = await octokit.repos.get({
-      owner: repoData.owner,
-      repo: repoData.repoName,
-    });
-  } catch (error) {
-    console.log(error);
+
+  const repo = await octokit.repos.get({
+    owner: repoData.owner,
+    repo: repoData.repoName,
+  });
+
+  const branch = await octokit.repos.getBranch({
+    owner: repoData.owner,
+    repo: repoData.repoName,
+    branch: input.branchName ? input.branchName : 'main',
+  });
+
+  if (!branch.url) {
+    throw new Error("There's no such branch for this repository");
```

Comment:
let|s add the branch name to the error message for easier debugging

Reply:
@pradel done.